### PR TITLE
Add iam: add support for API key

### PR DESCRIPTION
**Pull request by illi-j**
This PR adds support for IAM API key resources management.

### CODE REVIEW

**Comment by illi-j** [comment]
```
======================================== test session starts ========================================
platform darwin -- Python 3.7.6, pytest-5.3.2, py-1.8.1, pluggy-0.13.1 -- /usr/local/opt/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /Users/johnatan/Works/exoscale/python-exoscale, inifile: pytest.ini
collected 5 items

tests/test_iam.py::TestIAM::test_create_api_key
>>> tests/test_iam.py::TestIAM::test_create_api_key (setup): test started at 2020-01-14 15:12:14.354968
PASSED
<<< tests/test_iam.py::TestIAM::test_create_api_key (teardown): test stopped at 2020-01-14 15:12:14.662044
--- Test duration: 0.31s


tests/test_iam.py::TestIAM::test_list_api_key
>>> tests/test_iam.py::TestIAM::test_list_api_key (setup): test started at 2020-01-14 15:12:14.664326
PASSED
<<< tests/test_iam.py::TestIAM::test_list_api_key (teardown): test stopped at 2020-01-14 15:12:15.097818
--- Test duration: 0.43s


tests/test_iam.py::TestIAM::test_get_api_key
>>> tests/test_iam.py::TestIAM::test_get_api_key (setup): test started at 2020-01-14 15:12:15.251551
PASSED
<<< tests/test_iam.py::TestIAM::test_get_api_key (teardown): test stopped at 2020-01-14 15:12:15.744956
--- Test duration: 0.49s


tests/test_iam.py::TestIAM::test_list_api_key_operations
>>> tests/test_iam.py::TestIAM::test_list_api_key_operations (setup): test started at 2020-01-14 15:12:15.895133
PASSED
<<< tests/test_iam.py::TestIAM::test_list_api_key_operations (teardown): test stopped at 2020-01-14 15:12:16.078133
--- Test duration: 0.18s


tests/test_iam_apikey.py::TestIAMAPIKey::test_revoke
>>> tests/test_iam_apikey.py::TestIAMAPIKey::test_revoke (setup): test started at 2020-01-14 15:12:16.090853
PASSED
<<< tests/test_iam_apikey.py::TestIAMAPIKey::test_revoke (teardown): test stopped at 2020-01-14 15:12:16.611412
--- Test duration: 0.52s



========================================= 5 passed in 2.61s =========================================
```

File: exoscale/api/iam.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    ### API key
+
+    def create_api_key(self, name, operations=None):
+        """
+        Create an API key.
+
+        Parameters:
+            name (str): the API key name
+            operations (str): the API key description
```
Comment (falzm) [inline exoscale/api/iam.py]: The parameter description doesn't match the name.

File: exoscale/api/iam.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                yield APIKey._from_cs(self, i)
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
+
+    def get_api_key(self, key):
+        """
+        Get an API key.
+
+        Parameters:
+            key (str): the key of an API key
```
Comment (falzm) [inline exoscale/api/iam.py]:
```suggestion
            key (str): the API key unique identifier
```

File: tests/conftest.py [hunk 3]
```diff
@@ -389,3 +389,22 @@ def _page(name=None, teardown=True):
...
+
+        if teardown:
+            api_keys.append(api_key)
+
+        return api_key
+
+    yield _api_key
+
+    for api_key in api_keys:
+        res = exo.iam.cs.revokeApiKey(key=api_key["key"])
```
Comment (falzm) [inline tests/conftest.py]: Please ensure your files end with a newline (also applies for other new files in this PR).

File: exoscale/api/iam.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+
+@attr.s
+class APIKey(Resource):
+    """
+    An API key.
+
+    Attributes:
+        name (str): the API key display name
+        key (str): the API key
```
Comment (falzm) [inline exoscale/api/iam.py]:
```suggestion
        key (str): the API key unique identifier
```

File: exoscale/api/iam.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        self.operations = None
+        self.type = None
+
+
+class IamAPI(API):
+    """
+    An Exoscale IAM API client.
+
+    Parameters:
+        key (str): the IAM API key
```
Comment (falzm) [inline exoscale/api/iam.py]:
```suggestion
        key (str): the IAM API key unique identifier
```

File: exoscale/api/iam.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+                raise CloudStackApiException(e.error["errortext"], e.error)
+
+        return APIKey._from_cs(self, res)
+
+    def list_api_key_operations(self):
+        """
+        list all supported operations of an API key.
+
+       Returns:
+            [str] : list of operations for the current API key
```
Comment (falzm) [inline exoscale/api/iam.py]:
```suggestion
            [str]: list of operations for the current API key
```

File: exoscale/api/iam.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+class APIKey(Resource):
+    """
+    An API key.
+
+    Attributes:
+        name (str): the API key display name
+        key (str): the API key
+        type (str): the API key type
+        secret (str): the API key secret
+        operations ([str]): the API key operations
```
Comment (falzm) [inline exoscale/api/iam.py]:
```suggestion
        operations ([str]): a list of allowed API operations
```

**Comment by illi-j** [comment]
🤘

**Pull request merged**